api/src/test/java/org/openmrs/api/ObsServiceTest.java

```diff
@@ -1973,4 +1974,29 @@ public void shouldNotAutomaticallySetStatusWhenManuallyCopyingAnObs() throws Exc
 		assertThat(existing.getVoided(), is(true));
 		assertThat(newObs.getStatus(), is(Obs.Status.FINAL));
 	}
+
+	@Test
+	//@Verifies(value = "should add person with encounter patient for a new obs", method = "saveObs(Obs,String)")
+	public void saveObs_shouldUpdateObsPersonValueWithEncounterPatientForNewObs() throws Exception {
+		executeDataSet(ENCOUNTER_OBS_XML);
+
+		ObsService os = Context.getObsService();
+		ConceptService cs = Context.getConceptService();
+		EncounterService es = Context.getEncounterService();
+		PersonService ps = Context.getPersonService();
+
+		Obs o = new Obs();
+		o.setConcept(cs.getConcept(3));
+		o.setDateCreated(new Date());
+		o.setCreator(Context.getAuthenticatedUser());
+		o.setLocation(new Location(1));
+		o.setObsDatetime(new Date());
+		o.setValueText("Testing");
+		o.setPerson(ps.getPerson(2)); // Setting this person (2) as different from that of encounter person (7)
+		o.setEncounter(es.getEncounter(2));
```

Comment:
Can you please a sanity check that ensures that encounter.person is different from obs.person

Reply:
Ok I will work on it..Should I change all the attributes of one person object from other ?

Reply:
Not sure what you mean